iceoryx_posh/CMakeLists.txt

```diff
@@ -136,14 +136,16 @@ target_include_directories(iceoryx_posh
     source/version
     source/runtime
 )
-target_link_libraries(iceoryx_posh
+target_link_libraries(iceoryx_posh 
     PUBLIC
     iceoryx_utils::iceoryx_utils
+    #@todo - Is it required to link sanitizer libs here (Its already linked against iceoryx_utils)
+    ${SANITIZER_LINKER_FLAGS}        
```

Comment:
Is it required to link sanitizer libs  with iceoryx_posh (& other components as well) ?
(sanitizer libs are already linked against iceoryx_utils)
@elBoberido

Reply:
I would do so to be sure that the stacktrace from the sanitizer is complete. My guess is that otherwise the sanitizer cannot show the full path from posh to utils in the stacktrace.

You can add `-fsanitize=address` to the general compiler flags (`CFLAGS` and `CXXFLAGS`). See in CMakelists.txt in iceoryx_meta how it is done with the coverage flags. Then you can remove the `${SANITIZER_LINKER_FLAGS}` from the libraries.

Reply:
> You can add `-fsanitize=address` to the general compiler flags (`CFLAGS` and `CXXFLAGS`). See in CMakelists.txt in iceoryx_meta how it is done with the coverage flags. Then you can remove the `${SANITIZER_LINKER_FLAGS}` from the libraries.

I'm not sure about this. We deliberately moved away from using `CFLAGS` and `CXXFLAGS` and used `TARGET_COMPILE_OPTIONS` instead. Now we would revert those changes.

Reply:
Just one thing, `CUSTOM_FLAGS` might be too generic. Just in case iceoryx is part of a bigger framework which already use that variable. It would probably be better to have an iceoryx prefix.

`gcc_build_flags.cmake` is problematic, since this is also used for `clang` and `msvc` compiler ... and maybe in future for even more. I would just drop the `Detection` and then the name is quite generic.

Reply:
Then we name it like this `ICEORYX_FLAGS` in `build_flags.cmake`

Reply:
but it's not only build flags, it also set some cmake variables which are used at cmake configure time ;)

Reply:
Summarising the discussions (plus new points)

- Rename `IceoryxPlatformDetection.cmake` ->  `IceoryxPlatform.cmake` (as suggested by @elBoberido). This name reflects the purpose somehow. (or more near to describing the purpose :wink: :)
- Move `sanitize` related flags to this file

I would still prefer to use separate flags `SANITIZER_COMPILER_FLAGS` & `SANITIZER_LINKER_FLAGS` because

- `sanitize` option enabled for selected components (`iceoryx_utils` & `iceoryx_platform`)
- This will be expanded with additional flags in future  (other sanitizers)

Reply:
Regarding the name for the flags it should be named as instrumentation flags like they do it in GCC: [https://gcc.gnu.org/onlinedocs/gcc/Instrumentation-Options.html](https://gcc.gnu.org/onlinedocs/gcc/Instrumentation-Options.html) because sanitizer are only one aspect. There are other tools like gcov who needs also linker flags.
Proposal: INSTRUMENTATION_COMPILE_FLAGS and INSTRUMENTATION_LINKER_FLAGS

To your issue with the paths: You can try it with CMAKE_SOURCE_DIR which gives you the path to the toplevel cmake. In our case it is iceoryx_meta, which maybe can used as starting point. Keep in mind that this only work in standalone build of iceoryx.

Reply:
We could have a `ICEORYX_SANITIZER_FLAGS` and add them to the `ICEORYX_INSTRUMENTATION_FLAGS` so that we only need to add one variable to the target_compile_options and still have a separation

Reply:
I have reworked `sanitize` flags & defined `ICEORYX_SANITIZER_FLAGS`

However I doubt if we can have one global `ICEORYX_INSTRUMENTATION_FLAGS` (which includes `sanitize` & `coverage`), because `sanitize` & `coverage` has to be applied to targets separately 
e.g. `coverage` is applied to all the targets (including tests). However `sanitize` is applied to only core components.
Theoretically its possible to apply `sanitize` to all targets , however that will

- unnecessarily bloat the targets
- applies strict checks for all the components (including tests), which may leave the the **developers**  frustrated :wink:

Reply:
I'm a little bit torn by limiting it just to core targets. It might result in some technical dept in our tests, which we already have. It's okay for now, since fixing all the stuff in the tests would blow up this PR, but we should at least consider it.

Reply:
May be we can create a follow up issue to enable `sanitize` for all the targets ?

Reply:
@elBoberido Created #424 . Please add if I missed anything